mlflow/transformers.py

```diff
@@ -836,19 +852,37 @@ def _load_model(path: str, flavor_config, return_type: str, device=None, **kwarg
                 device = _TRANSFORMERS_DEFAULT_CPU_DEVICE_ID
         elif is_gpu_available():
             device = _TRANSFORMERS_DEFAULT_GPU_DEVICE_ID
+    # Note that we don't set the device in the conf yet because device is
+    # incompatible with device_map.
+    accelerate_model_conf = {}
+    if MLFLOW_HUGGINGFACE_USE_DEVICE_MAP.get():
+        device_map_strategy = MLFLOW_HUGGINGFACE_DEVICE_MAP_STRATEGY.get()
+        conf["device_map"] = device_map_strategy
+        accelerate_model_conf["device_map"] = device_map_strategy
+        # Cannot use device with device_map
+        device = None
 
-    local_path = pathlib.Path(path)
-    pipeline_path = local_path.joinpath(
-        flavor_config.get(_PIPELINE_BINARY_KEY, _PIPELINE_BINARY_FILE_NAME)
-    )
-
-    model_instance = getattr(transformers, flavor_config[_PIPELINE_MODEL_TYPE_KEY])
-    conf = {
-        "task": flavor_config[_TASK_KEY],
-        "model": model_instance.from_pretrained(pipeline_path),
-    }
     if device is not None:
         conf["device"] = device
+        accelerate_model_conf["device"] = device
+
+    if _TORCH_DTYPE_KEY in flavor_config:
+        dtype_val = _deserialize_torch_dtype_if_exists(flavor_config)
+        conf[_TORCH_DTYPE_KEY] = dtype_val
+        accelerate_model_conf[_TORCH_DTYPE_KEY] = dtype_val
+
+    accelerate_model_conf["low_cpu_mem_usage"] = MLFLOW_HUGGINGFACE_USE_LOW_CPU_MEM_USAGE.get()
+
+    if not MLFLOW_DISABLE_HUGGINGFACE_ACCELERATE_FEATURES.get():
+        try:
+            model = model_instance.from_pretrained(pipeline_path,
+                                                   **accelerate_model_conf)
+        except ValueError:
```

Comment:
When does `model_instance.from_pretrained` throw `ValueError`?

Reply:
Some model types don't support accelerate flags, and they throw ValueError when you pass parameters that don't exist

Reply:
Can you add that as a comment?

Reply:
Do TF PIpelines throw this or do they throw a Keras TypeError? We might want to wrap all of this accelerate logic around a validation of the underlying model type that was saved to ensure that we're not trying to include accelerate options on a Pipeline load for architectures that don't support it.

Reply:
Ah I see - I guess we could default this logic to false if the engine of the underlying model is TF?

Reply:
Yep I think that's fine. We'd have to load the `flavor_config` prior to executing `_load_model()` in order to get this logic to work and I feel as though that would be best served as a follow-up PR.